lab_5_scrapper/scrapper.py

```diff
@@ -13,70 +75,146 @@ def __init__(self, path_to_config: Path) -> None:
+
+        self._validate_config_content()
+
+        self._seed_urls = self.content['seed_urls']
+        self._num_articles = self.content['total_articles_to_find_and_parse']
+        self._headers = self.content['headers']
+        self._encoding = self.content['encoding']
+        self._timeout = self.content['timeout']
+        self._should_verify_certificate = self.content['should_verify_certificate']
+        self._headless_mode = self.content['headless_mode']
+
+        self.config_obj = self._extract_config_content()
 
     def _extract_config_content(self) -> ConfigDTO:
         """
         Returns config values
         """
-        pass
+        return ConfigDTO(**self.content)
 
     def _validate_config_content(self) -> None:
         """
         Ensure configuration parameters
         are not corrupt
         """
-        pass
+        if not isinstance(self.content['seed_urls'], list):
+            raise IncorrectSeedURLError
+
+        for element in self.content['seed_urls']:
+            if not isinstance(element, str):
+                raise IncorrectSeedURLError
+
+            if 'https://' not in element:
+                raise IncorrectSeedURLError
+
+        if not isinstance(self.content['total_articles_to_find_and_parse'], int)\
+                or isinstance(self.content['total_articles_to_find_and_parse'], bool)\
+                or self.content['total_articles_to_find_and_parse'] <= 0:
+            raise IncorrectNumberOfArticlesError(
+                "Total number of articles to parse is not integer"
+            )
+
+        if self.content['total_articles_to_find_and_parse'] > NUM_ARTICLES_UPPER_LIMIT:
+            raise NumberOfArticlesOutOfRangeError(
+                "Total number of articles is out of range from 1 to 150"
+            )
+
+        if not isinstance(self.content['headers'], dict):
+            raise IncorrectHeadersError("Headers are not in a form of dictionary")
+
+        if not isinstance(self.content['encoding'], str):
+            raise IncorrectEncodingError(
+                "Encoding must be specified as a string"
+            )
+
+        if not isinstance(self.content['timeout'], int):
+            raise IncorrectTimeoutError
+
+        if self.content['timeout'] <= TIMEOUT_LOWER_LIMIT\
+                or self.content['timeout'] > TIMEOUT_UPPER_LIMIT:
+            raise IncorrectTimeoutError(
+                "Timeout value must be a positive integer less than 60"
+            )
+
+        if not isinstance(self.content['headless_mode'], bool):
+            raise IncorrectVerifyError
+
+        if not isinstance(self.content['should_verify_certificate'], bool):
+            raise IncorrectVerifyError("Verify certificate value must either be True or False")
 
     def get_seed_urls(self) -> list[str]:
         """
         Retrieve seed urls
         """
-        pass
+        if isinstance(self._seed_urls, list):
+            return self._seed_urls
+        return ['']
```

Comment:
redundant check. getter should return part of the internal state. otherwise, you fool your user